war/src/main/js/add-item.js

```diff
@@ -125,22 +123,93 @@ $.when(getItems()).done(function(data){
     //////////////////////////
     // helper functions...
 
+    function setSelectState(e){
+      if(e.preventDefault){
+        e.preventDefault();
+      }
+      else{
+        e = {currentTarget:e};
+      }
+      var href = $(e.target).attr('href');
+      if(href) {
+        window.open(href);
+      }
+      
+      var $this = $(e.currentTarget).closest('li');
+      var $cat = $this.closest('.categories');
+      var $radios = $cat.find('input[type="radio"][name="mode"]').removeProp('checked');
+      var $checked = $radios.find(':checked');
+      
+      $cat.find('.active').removeClass('active');
+      
+      // Something in Jenkins land wants to reverse the radio button change on keydown. Waiting a pulse avoids the issue.
+      setTimeout(function(){
+        $this.addClass('active');
+        $this.find('input[type="radio"]').prop('checked', true);
+        checkFormReady();
+      },1);       
+    }    
+    
     function checkFormReady(){
+      
       //make sure everyone has changed and gotten attached...
       setTimeout(function(){
         var $name = $form.removeClass('no-select').find('input[name="name"]').removeClass('no-val');
+        var $nameBox = $name.closest('.j-add-item-name');
+        var val = $.trim($name.val());
+        var named = false;
+        
+        $nameBox.parent().find('.error').remove();
         
-        function checkItems(){
-          var selected = $form.find('input[type="radio"]:checked').length > 0;
-          var named = $.trim($name.val()).length > 0;
-          return {selected:selected,named:named};
+        function checkJobName(){
+          if(val.length === 0 ){
+            checkName('<div class="error">You must name this item.</div>');
+            return confirmBoth(false);
+          }
+          $.get('../../checkJobName?value='+val).always(checkName);
```

Comment:
Once #2192 is reviewed & merged, you should be able to use the relative URL `checkJobName?value=…`.